### ISSUE TITLE
Scoped Custom Element Registries

### ISSUE THREAD

**Issue by thescientist13** [WebKit/standards-positions#38]
## Request for position on an emerging web specification

* WebKittens who can provide input: @rniwa 

### Information about the spec

* Spec Title: Scoped Custom Element Registries
* Spec URL: https://github.com/whatwg/html/pull/10869
* Issue Tracker (if not the repository's issue tracker): https://github.com/whatwg/html/issues/10854

### Design reviews and vendor positions

* TAG Design Review: https://github.com/w3ctag/design-reviews/issues/1070
* Mozilla standards-positions issue: https://github.com/mozilla/standards-positions/issues/424

### Bugs tracking this feature

* WebKit Bugzilla: https://bugs.webkit.org/show_bug.cgi?id=267454
* Radar: rdar://121256394

### Anything else we need to know

Brief summary of the issue or the subject of the spec.
(taken from the OP)
> _The global registry is a possible source of name collisions that may arise from coincidence, or from **an app trying to define multiple versions of the same element**, or from more advanced scenarios like registering mocks during tests, or a component explicitly replacing an element definition for its scope._
> ...
> _This proposal adds the ability to construct `CustomElementRegistry`s and chain them in order to inherit custom element definitions. It uses ShadowRoot as a scope for definitions._

**Comment by annevk** [comment]
This is a bit tricky. Overall colleagues and I have been supportive of this effort, but since the proposal is still not finalized we cannot really provide a definitive position at this point. The details matter, after all. If our feedback continues to be taken into account it seems likely we'd end up supporting this though.

**Comment by keithamus** [comment]
https://bugs.webkit.org/show_bug.cgi?id=267454

**Comment by keithamus** [comment]
The position on this should likely be altered as Apple seem to be pursuing a design. /cc @rniwa @annevk

**Comment by annevk** [comment]
At this point colleagues and are solely interested in advancing https://github.com/whatwg/html/issues/10854 as detailed in its corresponding specification PRs.

This is different from the initial proposal, which I don't think is being advanced by anyone at this point. So we could either consider this a position for the revised proposal and update things accordingly or consider this as withdrawn. @thescientist13 any preference?

If we consider this as a position for the revised proposal I would suggest we resolve this as "position: support" one week from now.

**Comment by thescientist13** [comment]
Yeah, if it's just a matter of updating the spec links to point to the revised WHATWG link, happy to do that, as the WCCG seems in favor of maintaining this issue and just re-using it for the revised proposal.
